presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1567,6 +1575,46 @@ private Scope computeAndAssignOrderByScope(OrderBy node, Scope sourceScope, Scop
             return orderByScope;
         }
 
+        private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope outputScope, List<FunctionCall> aggregations, List<List<Expression>> groupByExpressions)
+        {
+            // This scope is only used for planning. When aggregation is present then
+            // only output fields, groups and aggregation expressions should be visible from ORDER BY expression
+            ImmutableList.Builder<Expression> orderByAggregationExpressionsBuilder = ImmutableList.builder();
+            if (groupByExpressions.size() == 1) {
+                orderByAggregationExpressionsBuilder.addAll(getOnlyElement(groupByExpressions));
+            }
+            orderByAggregationExpressionsBuilder.addAll(aggregations);
+
+            // Don't add aggregate expression that contains references to output column because the names would clash in TranslationMap during planning.
+            List<Expression> orderByAggregationExpressions = orderByAggregationExpressionsBuilder.build().stream()
+                    .filter(expression -> !hasEqualOrderByExpressionWithOutputColumnReference(expression, node, outputScope))
+                    .collect(toImmutableList());
+
+            // generate placeholder fields
+            ImmutableList.Builder<Field> orderByAggregationSourceFields = ImmutableList.builder();
+            orderByAggregationExpressions.forEach(expression -> orderByAggregationSourceFields.add(Field.newUnqualified(Optional.empty(), analysis.getType(expression))));
+
+            Scope orderByAggregationScope = Scope.builder()
+                    .withRelationType(new RelationType(orderByAggregationSourceFields.build()))
+                    .build();
+
+            Scope orderByScope = Scope.builder()
+                    .withParent(orderByAggregationScope)
+                    .withRelationType(outputScope.getRelationType())
+                    .build();
+            analysis.setScope(node, orderByScope);
+            analysis.setOrderByAggregates(node, orderByAggregationExpressions);
+            return orderByScope;
+        }
+
+        private boolean hasEqualOrderByExpressionWithOutputColumnReference(Node aggregation, OrderBy orderByNode, Scope outputScope)
```

Comment:
Why is `aggregation` of type `Node`? Shouldn't it be `Expression`?

Also, I think this would be cleaner and easier to reason about it first extracted expression from the OrderBy and then tried to identify those that are equal (or contain subexpressions that are equal) to the aggregation.